### ISSUE TITLE
The Shadow Range controls are unlabeled

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63899]
### Description

Discovered while testing https://github.com/WordPress/gutenberg/pull/63874

All the range controls in the Shadows popover are unlabeled. Screenshot:

![Screenshot 2024-07-24 at 13 59 45](https://github.com/user-attachments/assets/1621ebf6-e6f5-4ae9-afec-44398a734ba3)

### Step-by-step reproduction instructions

- Go to the Site editor > Styles > Shadows
- Edit one of the shadows.
- Inspect the source and observe all the 4 sliders are unlabeled: X Position, Y Position, Blur, Spread.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by afercia** [comment]
Introduced in https://github.com/WordPress/gutenberg/pull/60706

Additionally:
- The visible text: X Position, Y Position, Blur, Spread., are H2 headings. It would be best to use real, visible, label elements.
- Teh popover title [is a H5 heading](https://github.com/WordPress/gutenberg/blame/trunk/packages/edit-site/src/components/global-styles/shadows-edit-panel.js/#L426), which doesn't make sense in the headings hierarchy.

**Comment by afercia** [comment]
Noting that the design is inconsistent, as the toggle to switch the control to the input to enter a custom value is _always_ placed after the control in the editor. Instead, in this case it's placed after the title of the control, which is less than ideal.

Example of other controls:

![Screenshot 2024-07-24 at 14 36 42](https://github.com/user-attachments/assets/6eecd5e2-7b6f-49b1-8a72-1859d2b0c5be)

The best way to solve this issue is by changing the design, which would also make it consistent with other controls.

**Comment by afercia** [comment]
As far as I can tell, this is the only Popover that has a visible title. It's a H5 heading, which is an arbitrary level in the headings hierarchy. Most popovers don't have a visible title, e.g. the Color picker in the screenshot below. Sometimes they do have an `aria-label` but don't have an ARIA `role` so that's incorrect anyways. 
To fix the design and semantics inconsistency my preference would be to render a visible title for _all_ popovers and reference it with `aria-labelledby` to label the popover. Lastly, the popover should be a _non-modal_ `dialog`. I'll split this into a separate issue.

![Screenshot 2024-07-24 at 16 57 57](https://github.com/user-attachments/assets/2d7a24f9-8c85-4834-ac65-b70a88953ada)
